packages/aws-cdk-lib/aws-efs/lib/efs-file-system.ts

```diff
@@ -301,10 +318,75 @@ abstract class FileSystemBase extends Resource implements IFileSystem {
    * @param actions The actions to grant
    */
   public grant(grantee: iam.IGrantable, ...actions: string[]): iam.Grant {
-    return iam.Grant.addToPrincipal({
+    return iam.Grant.addToPrincipalOrResource({
+      grantee: grantee,
+      actions: actions,
+      resourceArns: [this.fileSystemArn],
+      resource: this,
+    });
+  }
+
+  /**
+   * Grant the client actions defined in actions to the given grantee on this File System resource.
+   * If this method is used and the allowAnonymousAccess props are not specified,
+   * anonymous access to this file system is prohibited.
+   *
+   * @param grantee The principal to grant right to
+   * @param actions The client actions to grant
+   * @param conditions The conditions to grant
+   * @internal
+   */
+  public _grantClient(grantee: iam.IGrantable, actions: ClientAction[], conditions?: Record<string, Record<string, unknown>>): iam.Grant {
```

Comment:
Does this need to be `public` or can it be `private`?

Reply:
I was planning to use this method on AccessPoint, so I made it public. Currently, it doesn't need to be public, so I will change to private until open the PR that add grant API to AccessPoint :)

Reply:
Incidentally, `_grantClient` is a method created to determine whether a client has been granted or not. If denying anonymous clients is the default behavior, then this method differs from `grant()` only by being able to specify conditions. If that's the case, I thought it would be okay to create a grant method that can specify conditions as `grantBeta1()`, but what do you think?